release-tools/check_plugin.sh

```diff
@@ -0,0 +1,196 @@
+  do
+    IFS=' '
+    plugin_name=`echo ${PLUGINS_LOCATION_ARRAY[$pindex]} | awk -F/ '{print $NF}'`
+    plugin_bucket=`echo ${PLUGINS_LOCATION_ARRAY[$pindex]} | awk -F/ '{print $3}'`
+    plugin_path=`echo ${PLUGINS_LOCATION_ARRAY[$pindex]} | sed "s/^.*$plugin_bucket\///g"`
+    plugin_type_array=( `echo ${PLUGINS_TYPE_ARRAY[$pindex]} | tr ',' ' '` )
+    plugin_keyword_array=( `echo ${PLUGINS_KEYWORD_ARRAY[$pindex]} | tr ',' ' '` )
+    plugin_total=$((plugin_total+${#plugin_type_array[@]}))
+
+    IFS=`echo -ne "\n\b"`
+
+    for tindex in ${!plugin_type_array[@]}
+    do
+      for kindex in ${!plugin_keyword_array[@]}
+      do
+        if [ "${plugin_keyword_array[$kindex]}" = "None" ]
+        then
+          plugin_latest=`aws s3api list-objects --bucket $plugin_bucket --prefix $plugin_path --query 'Contents[].[Key]' --output text \
+                         | grep $ODFE_VERSION | grep -i ${plugin_type_array[$tindex]} | sort | tail -n 1 | awk -F '/' '{print $NF}'`
+        else
+          plugin_latest=`aws s3api list-objects --bucket $plugin_bucket --prefix $plugin_path --query 'Contents[].[Key]' --output text \
+                         | grep $ODFE_VERSION | grep -i ${plugin_type_array[$tindex]} | grep -i ${plugin_keyword_array[$kindex]} | sort | tail -n 1 | awk -F '/' '{print $NF}'`
+        fi
+        if [ -z "$plugin_latest" ]
+        then
+          plugin_latest="unavailable:${plugin_type_array[$tindex]}:${plugin_name}"
+          unavailable_plugin+=( $plugin_latest )
+        else
+          plugin_latest="isavailable:${plugin_type_array[$tindex]}:${plugin_latest}"
+          available_plugin+=( $plugin_latest )
+        fi
+        echo $plugin_latest
+      done
+    done
+
+  done
+
+  cd $ROOT
+
+  echo "<h1><u>[$plugin_category] Plugins ($ODFE_VERSION) Availability Checks for ( ${#available_plugin[@]}/$plugin_total )</u></h1>" >> message.md
+  echo "[$plugin_category] Plugins ($ODFE_VERSION) for ( ${#available_plugin[@]}/$plugin_total ): " >> chime_message.md
+  
+  echo "<h2><p style='color:red;'><b>NOT AVAILABLE</b></p></h2>" >> message.md
+  if [ "${#unavailable_plugin[@]}" -gt 0 ]
+  then
+    RUN_STATUS=1
+    echo "<ol>" >> message.md
+    for item in ${unavailable_plugin[@]}
+    do
+      item=`echo $item | awk -F ':' '{print $NF}'`
+      echo "<li><h3>$item</h3></li>" >> message.md
+      echo ":x: $item " >> chime_message.md
+    done
+    echo "</ol>" >> message.md
+    echo "<br><br>" >> message.md
+  fi
+  
+#  echo "<h2><p style='color:gold;'><b>IN PROGRESS</b></p></h2>" >> message.md
+#  if [ "${#inprogress_plugin[*]}" -gt 0 ]
+#  then
+#    RUN_STATUS=1
+#    echo "<ol>" >> message.md
+#    for item in ${inprogress_plugin[*]}
+#    do
+#      item=`echo $item | awk -F ':' '{print $NF}'`
+#      echo "<li><h3>$item</h3></li>" >> message.md
+#      echo ":hourglass_flowing_sand: $item " >> chime_message.md
+#    done
+#    echo "</ol>" >> message.md
+#    echo "<br><br>" >> message.md
+#  fi
```

Comment:
Can you please remove the commented section.

Reply:
This is needed, as it will only work once another script is ready.
Will comment now and enable once the other script works.